google/api_core/exceptions.py

```diff
@@ -135,6 +143,19 @@ def errors(self):
         """
         return list(self._errors)
 
+    @property
+    def error_details(self):
```

Comment:
Is `details` property has been occupied?  If not, is that better to use `details` as name or `status_details` relate to `google.rpc.Status.details`. 
Notice that we need to keep consistency between grpc and REST.
IMO, I prefer `status_details`. WDYT @busunkim96 @tseaver @atulep